### PR TITLE
SSP 4.99.173

### PR DESCRIPTION
**Pull request by cPanelChrisD**
Revert one previous change, fix some minor issues found with other recent changes, "perltidy" everything again, and fix one annoying bug (TECH-409).

I've tested quite a few of the functions to be sure they are still working as intended, so it should be ready to go live after this is merged.

### CODE REVIEW

File: ssp [hunk 1]
```diff
@@ -8538,7 +8540,7 @@ END_OF_TEMPLATE
         if ( $pid == 0 ) {
             close $out_r;
 
-            open STDIN,  '</dev/null';
+            open STDIN,  '<',  '/dev/null';
```
Comment (davelcpanelnet-zz) [inline ssp:8543]: Why this change? - curiosity sake.

**Comment by davelcpanelnet-zz** [comment]
```

Looks good on 3 hosts I checked it out on.  Nothing to add - just a single question that's pretty fruitless.

File: ssp (hunk 1 above)
Comment (teodesian) [reply ssp:8543]: 2-arg open is generally frowned upon these days when not opening sockets.  Mostly because you can get in trouble with files that have operation characters in them (e.g. >,<).

Comment (cPanelChrisD) [reply ssp:8543]: perlcritic also forbids it (for the reason George mentioned) so 'make test' failed with:
`[InputOutput::ProhibitTwoArgOpen] Line 8543 Column 13 - Two-argument "open" used - See page 207 of PBP. (Severity: 5)`
